src/pages/SearchPage.js

```diff
@@ -41,68 +53,77 @@ const StyledSwitcher = styled(Switcher)`
 `};
 `
 
+function createSortingFunc(field) {
+  return (m1, m2) => {
+    const value1 = m1[field]
+    const value2 = m2[field]
+    if (value1 < value2) return -1
+    if (value2 < value1) return 1
+    return 0
+  }
+}
+
+const SORT = {
+  RATING: createSortingFunc('rating'),
+  RELEASE_DATE: createSortingFunc('release_year'),
+}
+
 class SearchPage extends Component {
-  constructor(props) {
-    super(props)
+  constructor() {
+    super()
 
     this.state = {
       sortType: Object.keys(SORT_TYPE)[0],
-      searchType: '',
-      searchText: '',
-      movies: [],
-      error: '',
     }
     this.handleTypeChange = this.handleTypeChange.bind(this)
-    this.preformSearch = this.preformSearch.bind(this)
-  }
-
-  handleTypeChange(sortType) {
-    this.setState({ sortType })
+    this.performSearch = this.performSearch.bind(this)
   }
 
   componentDidMount() {
     const query = this.props.match.params.query
     const { searchType, searchText } = urlToSearchParams(query)
-    this.preformSearch(searchType, searchText)
+    this.performSearch(searchType, searchText, true)
   }
 
-  preformSearch(type, text) {
-    const searchUrl = `/search/${paramsToUrl(type, text)}`
-    this.props.history.push(searchUrl)
+  componentDidUpdate(prevProps) {
+    const { query } = this.props.match.params
+    if (prevProps.match.params.query !== query) {
+      const { searchType, searchText } = urlToSearchParams(query)
+      this.performSearch(searchType, searchText, true)
+    }
+  }
+
+  handleTypeChange(sortType) {
+    this.setState({ sortType })
+  }
 
+  performSearch(type, text, doNotTouchHistory) {
     if (text.trim().length === 0) return
 
-    this.setState({ loading: true })
-    search(type, text).then(
-      movies => {
-        this.setState({
-          movies,
-          error: '',
-          loading: false,
-        })
-      },
-      error => {
-        this.setState({
-          error: error.message,
-          movies: [],
-          loading: false,
-        })
-      }
-    )
+    const searchUrl = `/search/${paramsToUrl(type, text)}`
+    this.props.setLoading('search', 'loading')
```

Comment:
It's better to use constants for such values (can be defined in a separate file)